libopendroneid/opendroneid.c

```diff
@@ -748,9 +756,17 @@ int decodeAuthMessage(ODID_Auth_data *outData, ODID_Auth_encoded *inEncoded)
         return ODID_FAIL;
 
     if (inEncoded->page_zero.DataPage == 0) {
-        if (inEncoded->page_zero.LastPageIndex >= ODID_AUTH_MAX_PAGES ||
-            inEncoded->page_zero.Length > MAX_AUTH_LENGTH)
+        if (inEncoded->page_zero.LastPageIndex >= ODID_AUTH_MAX_PAGES)
+            return ODID_FAIL;
+
+#if (ODID_AUTH_MAX_PAGES <= 11) 
+        if (inEncoded->page_zero.Length > MAX_AUTH_LENGTH)
+            return ODID_FAIL;
+#else
+        if (inEncoded->page_zero.LastPageIndex > 10 &&
+            inEncoded->page_zero.Length != UINT8_MAX)
```

Comment:
I would propose to fix this instead with the following code snippet:
```
#if (MAX_AUTH_LENGTH < UINT8_MAX)
        if (inEncoded->page_zero.Length > MAX_AUTH_LENGTH)
            return ODID_FAIL;
#endif
```

Reply:
Better. Done.